### PR TITLE
Add Request Lock to RequestWiki

### PR DESCRIPTION
**Pull request by pixDeVl**
## Summary by CodeRabbit

- **New Features**
  - Introduced a locking mechanism for wiki requests, allowing users to specify if a wiki page should be locked during creation.
  - Added a new checkbox field for locking functionality in the wiki creation form.

- **Bug Fixes**
  - Improved handling of request locking logic during form submission.

- **Database Changes**
  - Added a `cw_lock` column to the `cw_requests` table, enabling tracking of lock states for requests.

### CODE REVIEW

**Comment by RhinosF1** [comment]
@coderabbitai review

**Comment by RhinosF1** [comment]
@coderabbitai review

File: includes/RequestWiki/RequestWikiRequestViewer.php [hunk 1]
```diff
@@ -426,6 +432,7 @@
 				$request->decline( $formData['reason'], $user );
 			}
 		}
+		$request->setLock($formData['lock'])
```
[the remark this answers is not in the document]
Comment (RhinosF1) [reply includes/RequestWiki/RequestWikiRequestViewer.php]: @coderabbitai: this should be fixed.

File: sql/cw_requests.sql [hunk 2]
```diff
@@ -12,4 +12,5 @@ CREATE TABLE /*_*/cw_requests (
   cw_category VARCHAR(64) NOT NULL,
   cw_visibility TINYINT UNSIGNED NOT NULL DEFAULT '0',
   cw_bio TINYINT UNSIGNED NOT NULL DEFAULT '0'
+  cw_lock TINYINT NOT NULL DEFAULT '0'
```
Comment (redbluegreenhat) [inline sql/cw_requests.sql:15]: Doesn't match the SQL patch.
```suggestion
  cw_lock TINYINT UNSIGNED NOT NULL DEFAULT '0'
```

**Review by redbluegreenhat — CHANGES_REQUESTED** [review]

File: sql/cw_requests.sql [hunk 3]
```diff
@@ -12,4 +12,5 @@ CREATE TABLE /*_*/cw_requests (
   cw_category VARCHAR(64) NOT NULL,
   cw_visibility TINYINT UNSIGNED NOT NULL DEFAULT '0',
   cw_bio TINYINT UNSIGNED NOT NULL DEFAULT '0'
```
Comment (The-Voidwalker) [inline sql/cw_requests.sql:14]:
```suggestion
  cw_bio TINYINT UNSIGNED NOT NULL DEFAULT '0',
```

**Comment by Universal-Omega** [comment]
This will end up being done in #567

**Pull request closed without merging**